compiler/rustc_mir/src/transform/coverage/mod.rs

```diff
@@ -470,7 +470,7 @@ fn inject_statement(
             code_region: some_code_region,
         }),
     };
-    data.statements.push(statement);
+    data.statements.insert(0, statement);
```

Comment:
This is always going to be slower than appending at the end. I'd prefer to skip over coverage statements [in the elaborate drops pass](https://github.com/rust-lang/rust/blob/4031f7b0a8047653f2f989f2d8579bf5650e71ca/compiler/rustc_mir/src/dataflow/impls/mod.rs#L680-L717) rather than work around it here.

Reply:
I actually prefer it this way and I think the overall impact will be negligible.

Yes, functionally, we have the option of inserting the statement anywhere in the BCB, but it is standard practice to `insert()` into the `statements` vec in MIR processing when order is critical. 

As I was trying to say in my reply here (though maybe not clearly):

https://github.com/rust-lang/rust/issues/80045#issuecomment-745786169

I believe inserting the coverage statement between the last statement and the terminator is going to break up some sequences in unexpected ways. 

It wouldn't be a problem if I could insert the coverage statement **after** the terminator (which is impossible without a new BB, and that's much worse).

But the `Terminator` is really just a special case of a `Statement`, and the sequence from one statement to another is sometimes important, analytically at least. (Possibly even important at runtime, for context switching reasons, in fact.)

To drive the point: Think about a typical instruction sequence like, Load followed by Add (adding to the value that was just loaded). Most people would expect these to come in uninterrupted sequence, so injecting code to increment a global counter between the two, even if possible, would not be right.

I would rather not insert Coverage statements between a Statement and it's following Terminator, and it's not just about this one issue.

Reply:
I missed your comment there, makes sense.